internal/sirius/edit_investigation.go

```diff
@@ -24,12 +25,24 @@ func (c *Client) EditInvestigation(ctx Context, investigationID int, investigati
 	}
 	defer resp.Body.Close()
 
+	body, err := io.ReadAll(resp.Body)
+	if err != nil {
+		return err
+	}
+
 	if resp.StatusCode == http.StatusBadRequest {
-		var v SiriusValidationError
-		if err := json.NewDecoder(resp.Body).Decode(&v); err != nil {
+		var v ValidationError
+		var x FlexibleFields
+		if err := json.Unmarshal(body, &x); err != nil {
+			return err
+		}
+		formattedFieldErrors, err := formatToFieldErrors(x)
+		if err != nil {
 			return err
 		}
-		return FormatToValidationError(v)
+		v.Field = formattedFieldErrors
+		v.Detail = "Payload failed validation"
+		return v
```

Comment:
I originally only wanted to unmarshal the response body to `FlexibleFields` if it failed to decode the response body to the `ValidationError` type i.e. in the case that the error was coming from the handler. However, i kept getting an EOF error which i couldn't figure out..

```go
if resp.StatusCode == http.StatusBadRequest {
	var v ValidationError
	if err := json.NewDecoder(resp.Body).Decode(&v); err != nil {
		var x FlexibleFields
		if err := json.Unmarshal(body, &x); err != nil {
			return err
		}
		formattedFieldErrors, err := toStrict(x)
		if err != nil {
			return err
		}
		v.Field = formattedFieldErrors
		v.Detail = "Payload failed validation"
		return v
	}
	return v
}
```

Reply:
I think you can simplify this by using `FlexibleField` to replace `FieldErrors`, then here you could have:

```go
// removed the io.ReadAll here

if resp.StatusCode == http.StatusBadRequest {
	var v struct {
		Detail string              `json:"detail"`
		Field  flexibleFieldErrors `json:"validation_errors"`
	}
	if err := json.NewDecoder(resp.Body).Decode(&v); err != nil {
		return err
	}
	fieldErrors, err := v.Field.toFieldErrors()
	if err != nil {
		return err
	}
	return ValidationError{
		Detail: v.Detail,
		Field:  fieldErrors,
	}
}
```

I renamed `FlexibleFields` to `flexibleFieldErrors`, and then I've made the transformation function a method on that type, like:

```go
type flexibleFieldErrors map[string]json.RawMessage

func (f flexibleFieldErrors) toFieldErrors() (FieldErrors, error) {
	s := FieldErrors{}

	for k, v := range f {
		var asSlice []string
		if err := json.Unmarshal(v, &asSlice); err == nil {
			s[k] = map[string]string{"": strings.Join(asSlice, "")}
			continue
		}

		var asMap map[string]string
		if err := json.Unmarshal(v, &asMap); err == nil {
			s[k] = asMap
			continue
		}

		return nil, errors.New("could not parse field validation_errors")
	}

	return s, nil
}
```

and then you don't need to check for the `validation_errors` key.

Reply:
That's much cleaner, thanks i've made that change